app/code/Magento/InventoryCatalog/Model/Source/Validator/Bulk/DestinationSourceValidator.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Model\Source\Validator\Bulk;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\InventoryApi\Api\SourceRepositoryInterface;
+use Magento\InventoryCatalogApi\Api\DefaultSourceProviderInterface;
+use Magento\InventoryCatalogApi\Model\BulkInventoryTransferValidatorInterface;
+
+/**
+ * Check if sources exist and is available for transfer
+ * Do not perform any source/product cross check
+ */
+class DestinationSourceValidator implements BulkInventoryTransferValidatorInterface
+{
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+
+    /**
+     * @var SourceRepositoryInterface
+     */
+    private $sourceRepository;
+
+    /**
+     * @var DefaultSourceProviderInterface
+     */
+    private $defaultSourceProvider;
+
+    /**
+     * @param ValidationResultFactory $validationResultFactory
+     * @param SourceRepositoryInterface $sourceRepository
+     * @param DefaultSourceProviderInterface $defaultSourceProvider
+     * @SuppressWarnings(PHPMD.LongVariable)
+     */
+    public function __construct(
+        ValidationResultFactory $validationResultFactory,
+        SourceRepositoryInterface $sourceRepository,
+        DefaultSourceProviderInterface $defaultSourceProvider
+    ) {
+        $this->validationResultFactory = $validationResultFactory;
+        $this->sourceRepository = $sourceRepository;
+        $this->defaultSourceProvider = $defaultSourceProvider;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(array $skus, string $destinationSource, bool $defaultSourceOnly = false): ValidationResult
+    {
+        $errors = [];
+
+        if (($destinationSource === $this->defaultSourceProvider->getCode()) && $defaultSourceOnly) {
+            $errors[] = __('Cannot transfer default source to itself');
```

Comment:
why do you make a special case for Default Source, but don't make a similar check for custom sources? 

For example, that it's not possible to migrate from `Source A` -> `Source A` as this also does not make sense

Reply:
Because it is the only case in which you have a single and defined source, in the other cases you have all the sources as source.
But this will change with the new implementation discussed in the grooming session.